### PR TITLE
AuthError -> AuthenticationError/AuthorizationError

### LINKED ISSUE #80: Differentiate between 401 and 402 status codes
Currently we do not differentiate between 401 and 402 errors: https://github.com/twingly/twingly-search-api-ruby/blob/785d8604c2422eee7a1abc26422649fcdc57f861/lib/twingly/search/error.rb#L9-L10

Raising `AuthError` for 401 class of errors is fine, but for 402 we should probably raise something like `LanguageSelectionError`, see: https://developer.twingly.com/resources/search/#4xx-client-errors

### PR DESCRIPTION
**Pull request by jage**
~~Breaking change, removes AuthError exception from Search API~~

replacing AuthError with AuthenticationError (invalid API key) and AuthorizationError (denied access to a search feature). Both `AuthenticationError` and `AuthorizationError` inherits from `AuthError` to avoid breaking the API.

There's a slight inconsistency in the API since the API docs notes that
401 is "Unauthorized" when it should be "Unauthenticated":
https://developer.twingly.com/resources/search/#4xx-client-errors

Though this is a separate concern of the HTTP API.

Close #80

### CODE REVIEW

**Comment by jage** [comment]
Should we update LiveFeed once we're at it (also breaking)? LiveFeed don't have authorization error (no request options that can be stopped), so it would be a rename `AuthError` -> `AuthenticationError`.

File: lib/twingly/search/error.rb [hunk 1]
```diff
@@ -16,7 +18,10 @@ def self.from_api_response(code, message)
       end
     end
 
-    class AuthError < Error
+    class AuthenticationError < Error
```
Comment (roback) [inline lib/twingly/search/error.rb]: Just an idea: if we don't want any breaking changes we could keep `AuthError` around and have both `AuthenticationError` and `AuthorizationError` inherit it.

Comment (jage) [reply lib/twingly/search/error.rb]: Sure! That's easier to roll out, then maybe we could just leave `AuthError` in LiveFeed as is?

Comment (roback) [reply lib/twingly/search/error.rb]: Yes, sure. We can always add an `AuthenticationError` subclass to LiveFeed later if that is something we decide we want.

Comment (jage) [reply lib/twingly/search/error.rb]: @roback like 43c6388?

Comment (jage) [reply lib/twingly/search/error.rb]: If we think that's enough, I might as well add `AuthenticationError < AuthError` to LiveFeed as well.

Comment (roback) [reply lib/twingly/search/error.rb]: > @roback like [43c6388](https://github.com/twingly/twingly-search-api-ruby/commit/43c63887c90f8dc8b6e356df2fef56e2700c977d)?

:+1: Yes, that's exactly what I was thinking.

We can add it to LiveFeed now as well as that makes things less confusing for us/the users in the future.

Comment (jage) [reply lib/twingly/search/error.rb]: Yeah

File: spec/livefeed/parser_spec.rb [hunk 2]
```diff
@@ -306,6 +306,10 @@ module Twingly::LiveFeed
       context "with an unauthorized api key result" do
         let(:fixture) { :unauthorized_api_key }
 
+        it "should raise AuthenticationError" do
+          expect { subject }.to raise_error(AuthenticationError)
+        end
+
         it "should raise AuthError" do
```
Comment (roback) [inline spec/livefeed/parser_spec.rb]: I think we can remove this now. The spec was added above, and the spec added in `error_spec.rb` should be enough.

File: spec/livefeed/client_spec.rb [hunk 3]
```diff
@@ -42,6 +42,11 @@ module Twingly::LiveFeed
           expect { subject }
             .to raise_error(AuthError, "No API key has been provided.")
```
Comment (roback) [inline spec/livefeed/client_spec.rb]: Same with this one. I think we can remove it now.

File: spec/livefeed/parser_spec.rb (hunk 2 above)
Comment (jage) [reply spec/livefeed/parser_spec.rb]: Yeah, true. A bit too much overlap.

File: spec/livefeed/client_spec.rb (hunk 3 above)
Comment (jage) [reply spec/livefeed/client_spec.rb]: 71ce125

File: spec/livefeed/parser_spec.rb (hunk 2 above)
Comment (jage) [reply spec/livefeed/parser_spec.rb]: 71ce125

**Pull request merged**